src/test/incremental/hashes/function_interfaces.rs

```diff
@@ -284,7 +284,7 @@ pub fn change_return_impl_trait() -> impl Clone {
 }
 
 #[cfg(not(cfail1))]
-#[rustc_clean(cfg = "cfail2", except = "Hir, HirBody")]
+#[rustc_clean(cfg = "cfail2")]
```

Comment:
So this is the only change that I worry about. Not sure what's going on here. cc @michaelwoerister

Reply:
did you ever get resolution as to why this change is necessary and whether it is safe to do?

Reply:
I think the `Hir` is clean, because we now store just a `NodeId` and a `DefId` to the `existential type` declaration. If you change the trait of the `impl Trait`, those ids won't change. If changing the type of a type alias has the same effect on the incremental cache, then this change is fine, if not, then it's probably not fine.